token-client-core/src/main/java/no/nav/security/token/support/client/core/ClientProperties.java

```diff
@@ -17,35 +20,39 @@
 
     private static final List<OAuth2GrantType> GRANT_TYPES = List.of(
         OAuth2GrantType.JWT_BEARER,
-        OAuth2GrantType.CLIENT_CREDENTIALS
+        OAuth2GrantType.CLIENT_CREDENTIALS,
+        OAuth2GrantType.TOKEN_EXCHANGE
     );
 
     @NotNull
     private final URI tokenEndpointUrl;
     @NotNull
     private final OAuth2GrantType grantType;
-    @NotEmpty
     private final List<String> scope;
     @NotNull
     private final ClientAuthenticationProperties authentication;
     private final URI resourceUrl;
+    private final ExchangeProperties tokenExchange;
 
     public ClientProperties(@NotNull URI tokenEndpointUrl,
                             @NotNull OAuth2GrantType grantType,
-                            @NotEmpty List<String> scope,
+                            List<String> scope,
```

Comment:
Not sure about this. Does this break expected behaviour? I assume we've dropped this as token-exchange does not use scopes, however this might/should be required for the other methods?

Reply:
yeah, was not complitly sure about this, could drag it in to ExchangeProperties instead and keep ClientProperties as is, or we could just drop it..?

Reply:
for now.

Reply:
fixing :D